salvo/src/lib/source_tree.py

```diff
@@ -127,15 +160,69 @@ def pull(self) -> bool:
     Returns:
       a boolean indicating whether the operation was successful
     """
-    return False
+
+    source_name = proto_source.SourceRepository.SourceIdentity.Name(
+        self._source_repo.identity)
+    log.debug(f"Pulling [{source_name}] from origin: [{self._source_url}]")
+    self._validate()
```

Comment:
I can move the validate call, though I don't think that getting the source identity name from the value would affect things.

Reply:
Understood, this was mostly a question because the unusual placement (compared to the other methods) implied that this has some semantic meaning. Thanks for unifying it.